pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -15,9 +15,39 @@ rec {
         --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain"
     '';
 
-    phases = "unpackPhase installPhase";
+    fixupPhase =
+      let
+        arch =
+          if stdenv.system == "x86_64-linux" then "linux-amd64"
+          else if stdenv.system == "i686-linux" then "linux-i386"
+          else "";
+       in
+    ''
+      if [ -n "${arch}" ]; then
+        mkdir patching
+        pushd patching
+        jar xf $out/lib/gradle/lib/native-platform-${arch}-0.10.jar
+        patchelf --set-rpath "${stdenv.cc.cc}/lib:${stdenv.cc.cc}/lib64" net/rubygrapefruit/platform/${arch}/libnative-platform.so
+        jar cf native-platform-${arch}-0.10.jar .
+        mv native-platform-${arch}-0.10.jar $out/lib/gradle/lib/
+        popd
 
-    buildInputs = [ unzip jdk makeWrapper ];
+        # The scanner doesn't pick up the runtime dependency in the jar.
+        # Manually add a reference where it will be found.
+        echo ${stdenv.cc.cc} > $out/nix_references
+      fi
+    '';
+
+    phases = "unpackPhase installPhase fixupPhase";
```

Comment:
Style: if you're only trying to avoid a specific phase, I prefer to disable it directly, rather than list all the other phases. It looks like you could use `dontBuild = true` or `buildPhase = ":"` to achieve the same effect here.